### PR TITLE
[Feature][plugin] Add snowflake datasource in datasource plugin

### LINKED ISSUE #13727: [Feature][plugin] Support Snowflake datasource
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Snowflake is a really popular data warehouse based on cloud. Thus, we would like to enable Snowflake datasource in DS to support more use cases.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhaohehuhu**
## Purpose of the pull request
close  https://github.com/apache/dolphinscheduler/issues/13727

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Please check the failed CI.

**Comment by fuchanghai** [comment]
plz refer to #13727

**Comment by zhaohehuhu** [comment]
> plz refer to #13727

got it. will check later

**Comment by fuchanghai** [comment]
> > plz refer to #13727
> 
> got it. will check later

sorry ， l mean that this PR should be associated with issue

**Comment by zhaohehuhu** [comment]
> > > plz refer to #13727
> > 
> > 
> > got it. will check later
> 
> sorry ， l mean that this PR should be associated with issue

Done

**Comment by zhaohehuhu** [comment]
Hi All, what should I do for this ? @EricGao888 @SbloodyS @fuchanghai

**Comment by fuchanghai** [comment]
Hi @zhaohehuhu   plz use ``` -``` instead of ``` :```
image

**Comment by zhaohehuhu** [comment]
> Hi @zhaohehuhu plz use ` -` instead of ` :` image

Thanks. Already done

**Comment by zhaohehuhu** [comment]
> > Hi @zhaohehuhu plz use ` -` instead of ` :` image
> 
> Thanks. Already done

@EricGao888 @SbloodyS @fuchanghai Hi all, is there some things left?

**Comment by zhaohehuhu** [comment]
Hi all, I added a few UTs inside. Please help check if I need to do something else. @EricGao888 @fuchanghai @SbloodyS

**Comment by fuchanghai** [comment]
@SbloodyS plz run CI

File: dolphinscheduler-bom/pom.xml [hunk 1]
```diff
@@ -105,7 +105,10 @@
         <aws-java-sdk-redshift.version>1.12.300</aws-java-sdk-redshift.version>
         <azure-resourcemanager.version>2.21.0</azure-resourcemanager.version>
         <azure-resourcemanager-datafactory.version>1.0.0-beta.19</azure-resourcemanager-datafactory.version>
+        <snowflake-jdbc.version>3.13.10</snowflake-jdbc.version>
         <google-cloud-storage.version>2.18.0</google-cloud-storage.version>
+        <snowflake-jdbc.version>3.13.10</snowflake-jdbc.version>
+
```
Comment (Radeity) [inline dolphinscheduler-bom/pom.xml:111]: Duplicated.

Comment (Radeity) [inline dolphinscheduler-bom/pom.xml]: Duplicated.

**Comment by zhaohehuhu** [comment]
@SbloodyS plz help run CI

**Comment by zhaohehuhu** [comment]
@SbloodyS plz help run CI

**Comment by fuchanghai** [comment]
@zhaohehuhu plz run 'mvn spotless:apply'  to format codes

**Comment by zhaohehuhu** [comment]
> plz help run CI



> @zhaohehuhu plz run 'mvn spotless:apply' to format codes

done. and plz help run CI.

**Comment by zhaohehuhu** [comment]
@fuchanghai @SbloodyS @Radeity Is there anything that need to improve ? Plz help

**Comment by Radeity** [comment]
Basically LGTM, can you leave a blank space on the right of `snowflake` in Chinese doc?
> 数据源：选择 SNOWFLAKE 数据源

**Review by fuchanghai — DISMISSED** [review]
Basically LGTM，on my side

**Comment by zhaohehuhu** [comment]
> SNOWFLAKE

done. Thanks

**Comment by zhaohehuhu** [comment]
> Basically LGTM，on my side

Thanks

**Comment by fuchanghai** [comment]
@zhaohehuhu plz resolve conflicts

**Comment by zhaohehuhu** [comment]
> @zhaohehuhu plz resolve conflicts
done.

**Comment by zhaohehuhu** [comment]
@fuchanghai @Radeity @SbloodyS  Is it possible to merge this PR now?  Thanks.

**Comment by zhaohehuhu** [comment]
@Radeity @Radeity @SbloodyS  Is there anything that needs to get improved ?

**Review by SbloodyS — DISMISSED** [review]
LGTM.

**Comment by zhongjiajie** [comment]
@zhaohehuhu do you mind solving the conflict? and we will merge ASAP after that

**Comment by zhaohehuhu** [comment]
> @zhaohehuhu do you mind solving the conflict? and we will merge ASAP after that

done. Plz merge later

**Comment by zhongjiajie** [comment]
approval CI

**Comment by zhongjiajie** [comment]
restart the failed CI

**Review by zhongjiajie — APPROVED** [review]
Thanks 🎉

**Pull request merged**